rackcity/views/asset_views.py

```diff
@@ -401,8 +533,40 @@ def asset_modify(request):
             status=HTTPStatus.BAD_REQUEST
         )
     id = data['id']
+    if request.query_params.get("change_plan"):
+        change_plan = get_change_plan(request.query_params.get("change_plan"))
+        del data['id']
+    else:
+        change_plan = None
     try:
-        existing_asset = Asset.objects.get(id=id)
+        if change_plan:
+            assets, assets_cp = get_assets_for_cp(change_plan.id)
+            if assets_cp.filter(id=id).exists():
+                #if asset was created on a change_plan
+                existing_asset = assets_cp.get(id=id)
+            
+            else:
+                existing_asset_master = assets.get(id=id)
+                ## if asset has been modified before and exists on cp 
+                if AssetCP.objects.filter(
+                        change_plan=change_plan,
+                        related_asset=existing_asset_master).exists():
+                    existing_asset = AssetCP.objects.get(              
+                        change_plan=change_plan, 
+                        related_asset=existing_asset_master)
```

Comment:
can't you just do `assets_cp.get(id=id)` in both of these cases? aren't there only 2 cases? 
to make sure i'm following your logic correctly - you're just trying to get the existing asset in this part of the code. so first check if it's in the CP table, if it's not, then you need to get it from the Asset table and copy it to the Asset table?